components/perl/File-Listing/Makefile

```diff
@@ -22,30 +22,53 @@
 #
 # Copyright (c) 2011, 2013, Oracle and/or its affiliates. All rights reserved.
 # Copyright (c) 2014, Alexander Pyhalov. All rights reserved.
+# Copyright (c) 2022 Tim Mooney.  All rights reserved.
 #
 
+BUILD_STYLE=makemaker
+BUILD_BITS=32_and_64
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		File-Listing
 COMPONENT_VERSION=	6.04
-COMPONENT_REVISION=	1
+COMPONENT_REVISION=	2
 IPS_COMPONENT_VERSION=	6.4
+COMPONENT_FMRI=		library/perl-5/file-listing
+COMPONENT_SUMMARY=	File::Listing - parse directory listing
+COMPONENT_CLASSIFICATION=Development/Perl
 COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
 COMPONENT_ARCHIVE=	$(COMPONENT_SRC).tar.gz
-COMPONENT_PROJECT_URL=	http://search.cpan.org/~gaas/File-Listing/
+COMPONENT_PROJECT_URL=	https://metacpan.org/pod/File::Listing
 COMPONENT_ARCHIVE_HASH=	\
     sha256:1e0050fcd6789a2179ec0db282bf1e90fb92be35d1171588bd9c47d52d959cf5
-COMPONENT_ARCHIVE_URL=	http://search.cpan.org/CPAN/authors/id/G/GA/GAAS/$(COMPONENT_ARCHIVE)
+COMPONENT_ARCHIVE_URL=	https://cpan.metacpan.org/authors/id/P/PL/PLICEASE/$(COMPONENT_ARCHIVE)
+COMPONENT_LICENSE=	Artistic
 
-include $(WS_TOP)/make-rules/prep.mk
-include $(WS_TOP)/make-rules/makemaker.mk
-include $(WS_TOP)/make-rules/ips.mk
+# until all perl modules have been rebuilt with 5.34, each module sets
+# this manually, before including common.mk
+PERL_VERSIONS = 5.22 5.24 5.34
+PERL_64_ONLY_VERSIONS = 5.24 5.34
 
+include $(WS_MAKE_RULES)/common.mk
 
-COMPONENT_TEST_TARGETS = test
+#
+# use the same results for comparison for all builds
+#
+COMPONENT_TEST_MASTER = $(COMPONENT_TEST_RESULTS_DIR)/results-all.master
 
-build:		$(BUILD_32_and_64)
+#
+# filter out all output up to and including the test_harness line
+# filter out timing information and anything else that varies between builds
+#
+COMPONENT_TEST_TRANSFORMS += \
+	'-e "1,/test_harness/d"' \
+	'-e "s/,  *[0-9]* wallclock.*//"'
 
-install:	$(INSTALL_32_and_64)
+# because of the runtime dependency on http-date, it must be installed
+# at build time to run the tests
+REQUIRED_PACKAGES += library/perl-5/http-date
```

Comment:
I think the correct way is to have dependencies on real packages not the meta package:
REQUIRED_PACKAGES += library/perl-5/http-date-522
REQUIRED_PACKAGES += library/perl-5/http-date-524
REQUIRED_PACKAGES += library/perl-5/http-date-534

Reply:
I think that meta packages are for convenience purposes of humans only.
If you use meta packages for dependency management you are prone to errors, eg. you won't detect if somebody removes package versions from a meta package. I detected a real problem for the ansible package in the past. Ansible was dependent only on meta packages (python in this case). When we changed ansible's python version to 3.7 lots of dependent packages weren't available for python 3.7 and this stayed undetected by the build and runtime dependency checks.

Reply:
Ok, I'll switch to using versioned dependencies in the `Makefile` for build dependencies.

If I go back and fix the couple of modules where I've already used the meta-package for the build dependency and submit a new PR, do I need to increase `COMPONENT_REVISION` for those packages, or is changing just the build dependency "safe" and we don't need to trigger a package rebuild?

Reply:
Every build needs a change in the package's version information. The version information has different parts:
COMPONENT_VERSION (or more precisely IPS_COMPONENT_VERSION) +
RELEASE_MAJOR
RELEASE_MINOR
COMPONENT_REVISION
(look at ips-buildinfo.mk for their definitions).
These defintions are hierarchical, that means if a definition higher up changes you can reset the definitions below.
Beginning of 2022 I changed the definition of RELEASE_MAJOR to 2022. That makes it possible to reset RELEASE_MINOR and COMPONENT_REVISION to 0, but only one time for each package.
After that you either need a new IPS_COMPONENT_VERSION or a new COMPONENT_REVISION (RELEASE_MAJOR and RELEASE_MINOR can be changed but this happens only for certain circumstances).

So, yes you'll need to increment the COMPONENT_REVISION for every package that have already been touched this year, and you can reset COMPONENT_REVISION for every package you change the COMPONENT_VERSION or that haven't been touched recently.